portfolio/src/main/webapp/script.js

```diff
@@ -32,3 +32,16 @@ async function getMessageFromServer() {
   const quote = await response.text();
   document.getElementById('message-container').innerHTML = quote;
 }
+
+function login(url){
+  window.location = url;
+}
+
+async function setButtonName(){
```

Comment:
Consider renaming to configureLoginLink() or something since you're not just setting a button name.